zigbee:ncp: Implement osif bootloader functions (#4751)
nrfconnect/sdk-nrf: samples/zigbee/ncp/README.rst

To boot to the bootloader on the dongle, an additional action is required.
For example, you can trigger the pin reset procedure to reboot the device to the bootloader DFU mode, in which the new application image can be uploaded.

> See `nRF5 Bootloader DFU Mode`_ for the list of possible conditions for activating the DFU mode. The DFU bootloader can also be triggered by the call to
> ncp_host_ota_run_bootloader in the ncp host application.

Reviewer: ```suggestion
See `nRF5 Bootloader DFU Mode`_ for the list of possible conditions for activating the DFU mode.
You can also activate the bootloader DFU mode with a call to :c:func:`ncp_host_ota_run_bootloader` in the NCP Host application.
```
Few questions:
- Where is this function? I don't see it referenced in the main.c you've edited in this PR.
- Does this new line about call to the function in the NCP Host application refers to triggering the bootloader on the dongle?
- What changes in the code I must make to ensure this function is called?

Reviewer: Also, I see this is on the OSIF side. Shouldn't this be documented in the OSIF library page, @tomchy ?

Reviewer 2: This function is available only on the host side. If you call it, it is expected that the SoC will enter bootloader.

Author: > Where is this function? I don't see it referenced in the main.c you've edited in this PR.

This function is in ncp host package.

> Does this new line about call to the function in the NCP Host application refers to triggering the bootloader on the dongle?

Yes, this will trigger the bootloader on the dongle (assuming nRF5 SDK Bootloader is there). If this function is used with dk, which uses MCUboot, it will do nothing, because it goes to bootloader during each startup.

> What changes in the code I must make to ensure this function is called?
This is called everytime ncp soc receives the command to trigget bootloader.
